modules/rte_mis_student/rte_mis_student.module

```diff
@@ -327,6 +327,35 @@ function rte_mis_student_validate_mini_node_student($form, FormStateInterface $f
   if (isset($values['field_gender'][0]['value']) && $values['field_gender'][0]['value'] == 'boy' && isset($values['field_single_girl_child'][0]['value']) && $values['field_single_girl_child'][0]['value'] == 1) {
     $form_state->setErrorByName('field_single_girl_child', t('Single girl child with gender boy is invalid choice. Please select other gender or `No` in Single girl child'));
   }
+  // Save only the `field_parent_type` related values.
+  if (isset($values['field_parent_type'][0]['value'])) {
+    if ($values['field_parent_type'][0]['value'] == 'father_mother') {
+      $form_state->setValue('field_single_parent_type', ['0' => ['value' => NULL]]);
+      $form_state->setValue('field_guardian_name', ['0' => ['value' => '']]);
+      $form_state->setValue('field_gaurdian_aadhar_number', ['0' => ['value' => '']]);
+    }
+    elseif ($values['field_parent_type'][0]['value'] == 'single_parent') {
+      $form_state->setValue('field_guardian_name', ['0' => ['value' => '']]);
+      $form_state->setValue('field_gaurdian_aadhar_number', ['0' => ['value' => '']]);
+      if (isset($values['field_single_parent_type'])) {
+        if ($values['field_single_parent_type'][0]['value'] == 'father') {
+          $form_state->setValue('field_mother_name', ['0' => ['value' => '']]);
+          $form_state->setValue('field_mother_aadhar_number', ['0' => ['value' => '']]);
+        }
+        else {
+          $form_state->setValue('field_father_name', ['0' => ['value' => '']]);
+          $form_state->setValue('field_father_aadhar_number', ['0' => ['value' => '']]);
+        }
+      }
+    }
+    elseif ($values['field_parent_type'][0]['value'] == 'guardian') {
+      $form_state->setValue('field_single_parent_type', ['0' => ['value' => NULL]]);
+      $form_state->setValue('field_father_name', ['0' => ['value' => '']]);
+      $form_state->setValue('field_father_aadhar_number', ['0' => ['value' => '']]);
+      $form_state->setValue('field_mother_name', ['0' => ['value' => '']]);
+      $form_state->setValue('field_mother_aadhar_number', ['0' => ['value' => '']]);
+    }
+  }
```

Comment:
Can we use `form_state->getValues` at once unset the values in an array and then assign the values using `form_state->setValues`?

Reply:
ok